python/cudf/cudf/core/scalar.py

```diff
@@ -149,7 +149,7 @@ def __neg__(self):
         return self._scalar_unaop("__neg__")
 
     def __repr__(self):
-        return f"Scalar({self.value}, dtype={self.dtype})"
+        return f"Scalar({str(self.value)}, dtype={self.dtype})"
```

Comment:
Why do we need to call `str(self.value)` in an f-string?

Reply:
There's something wrong with subbing certain instances of numpy timedeltas into f-strings:

```
In [5]: import numpy as np
In [6]: x = np.timedelta64('NaT', 'ns')
In [7]: f"{x}"
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-7-10bea4cf1a38> in <module>
----> 1 f"{x}"

TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```

Reply:
That sounds like a bug we should raise on the numpy repo. Good find.

Can you just add `#TODO` comment with the issue number here?

Reply:
Raised https://github.com/numpy/numpy/issues/17552